corehq/apps/userreports/tests/test_report_builder.py

```diff
@@ -165,6 +169,29 @@ def test_owner_as_location(self):
             owner_location_prop_archived_w_descendants.get_text()
         )
 
+    def test_builder_for_registry(self):
+        case_type_for_registry = CaseType(domain=self.domain, name='registry_prop', fully_generated=True)
+        case_type_for_registry.save()
+        CaseProperty(case_type=case_type_for_registry, name='registry_property',
+                     deprecated=False, data_type='plain', group='').save()
+        user = create_user("admin", "123")
+        registry = create_registry_for_test(user, self.domain, invitations=[
+            Invitation('foo', accepted=True), Invitation('user-reports', accepted=True),
+        ], name='registry')
+        registry_data_source = get_sample_registry_data_source(registry_slug=registry.slug)
+        registry_data_source.save()
+
+        builder = RegistryCaseDataSourceHelper(self.domain, registry.slug, 'case', case_type_for_registry.name)
+
+        expected_property_names = ['closed', 'closed_on', 'registry_property', 'computed/owner_name',
+                                   'computed/user_name']
+        self.assertEqual(expected_property_names, list(builder.data_source_properties.keys()))
+        registry_prop = builder.data_source_properties['registry_property']
+        self.assertEqual('registry_property', registry_prop.get_id())
+        self.assertEqual('registry property', registry_prop.get_text())
+        case_type_for_registry.delete()
+        registry.delete()
```

Comment:
you shouldn't need to delete SQL models in tests since they get removed when the test transaction is rolled back. However if you did want to the proper way is to use `self.addCleanup(registry.delete)` immediately after you've created it. This will get called during test tear down.

Reply:
sweet thank you